src/processes/follow-topic-process.js

```diff
@@ -8,31 +8,33 @@ const {
   ChatAnonUserInfo
 } = require('../databases/models');
 const BetterSocialConstantListUtils = require('../utils/constantList/utils');
-const CHANNEL_TYPE_ANONYMOUS = require('../utils/constant');
+const {CHANNEL_TYPE_ANONYMOUS} = require('../utils/constant');
 const updateBetterChannelMembers = require('../getstream/updateBetterChannelMembers');
 
-const generate_channel_id_for_anon_chat = (owner, member, context = null, source_id = null) => {
+const generate_channel_id_for_anon_chat = (owner, member, context = null, sourceId = null) => {
   const hash = crypto.createHash('sha256');
   hash.update(owner);
   hash.update(member);
   if (context) hash.update(context);
-  if (source_id) hash.update(source_id);
+  if (sourceId) hash.update(sourceId);
 
   return hash.digest('hex');
 };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `generate_channel_id_for_anon_chat` uses a cryptographic hash function to generate a unique channel id. However, it's important to note that while the chance is extremely low, SHA-256 can still produce collisions. If two different inputs produce the same hash, this could lead to unexpected behavior in your application. Consider adding a mechanism to handle potential hash collisions.
```diff
const generate_channel_id_for_anon_chat = (owner, member, context = null, sourceId = null) => {
   const hash = crypto.createHash('sha256');
   hash.update(owner);
   hash.update(member);
   if (context) hash.update(context);
  if (sourceId) hash.update(sourceId);
 
   return hash.digest('hex');
 };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->